core/src/test/java/io/questdb/test/cairo/pool/ReaderPoolTest.java

```diff
@@ -268,6 +266,114 @@ public void testCloseWithInactiveReader() throws Exception {
         });
     }
 
+    @Test
+    public void testConcurrentGetAndGetCopyOf() throws Exception {
+        final int readerThreadCount = 4;
+        final int readerIterations = 200;
+        final int writerIterations = 50;
+        final int writerBatchSize = 10;
+        final String tableName = "test";
+
+        assertWithPool((ReaderPool pool) -> {
+            TableModel model = new TableModel(configuration, tableName, PartitionBy.HOUR)
+                    .col("sym", ColumnType.SYMBOL)
+                    .timestamp("ts");
+            AbstractCairoTest.create(model);
+
+            final CyclicBarrier barrier = new CyclicBarrier(readerThreadCount + 1);
+            final CountDownLatch halt = new CountDownLatch(readerThreadCount + 1);
+            final AtomicInteger errors = new AtomicInteger();
+
+            new Thread(() -> {
+                final Rnd rnd = new Rnd();
```

Comment:
these are pseudo-random, lets make it into a fuzz test with reproducible seeds

Reply:
Done in a01a26d2314e8e2f2f766431a77e355fae147b62